api/bots/github_detail/github_detail.py

```diff
@@ -0,0 +1,116 @@
+import re
+import os
+import sys
+import logging
+import six.moves.configparser
+
+import requests
+
+class GithubHandler(object):
+    '''
+    This bot provides details on github issues and pull requests when they're
+    referenced in the chat.
+    '''
+
+    CONFIG_PATH = os.path.expanduser('~/github_detail.ini')
+    GITHUB_ISSUE_URL_TEMPLATE = 'https://api.github.com/repos/{owner}/{repo}/issues/{id}'
+    GITHUB_PULL_URL_TEMPLATE = 'https://api.github.com/repos/{owner}/{repo}/pulls/{id}'
+    HANDLE_MESSAGE_REGEX = re.compile("(?:([\w-]+)\/)?([\w-]+)?#(\d+)")
+    MAX_LENGTH_OF_MESSAGE = 200
+
+    def __init__(self):
+        config = six.moves.configparser.ConfigParser()
+        with open(self.CONFIG_PATH) as config_file:
+            config.readfp(config_file)
+        if config.get('GitHub', 'owner'):
+            self.owner = config.get('GitHub', 'owner')
+        else:
+            # Allowing undefined default repos would require multiple triage_message regexs.
+            # It's simpler to require them to be defined.
+            sys.exit('Default owner not defined')
+
+        if config.get('GitHub', 'repo'):
+            self.repo = config.get('GitHub', 'repo')
+        else:
+            sys.exit('Default repo not defined')
+
+    def usage(self):
+        # type: () -> None
+        return ("This plugin displays details on github issues and pull requests. "
+                "To reference an issue or pull request usename mention the bot then "
+                "anytime in the message type its id, for example:\n"
```

Comment:
Here it would probably be s/anytime/anywhere, but maybe we can even skip it. What about:

> To reference an issue or pull request, at-mention the bot and type either its ID (`#1234`), the repository and the ID (`zulip#1234`), or the owner, the repository and the ID (`zulip/zulip#1234`).
> <examples>

Also, it'd be great to have different examples for the different types of use. Having all of them in the same query to the bot is a bit confusing.

Looks like you can ask for multiple issues in the same message. Maybe it's worth documenting that as a feature as well?